lib/jira/client/axios.js

```diff
@@ -1,86 +1,61 @@
 const axios = require('axios')
 const jwt = require('atlassian-jwt')
+const querystring = require('querystring')
 const url = require('url')
 
 const { logger } = require('probot/lib/logger')
 
 const instance = process.env.INSTANCE_NAME
 const iss = 'com.github.integration' + (instance ? `.${instance}` : '')
 
+/*
+ * Create an Atlassian request from an Axios config.
+ * Compatible with jwt.createQueryStringHash.
+ */
+jwt.reqFromAxiosConfig = (config) => {
+  const parsedUrl = url.parse(config.url)
+  const parsedQuery = querystring.parse(parsedUrl.query)
+
+  return {
+    method: config.method,
+    originalUrl: parsedUrl.pathname,
+    query: Object.assign(parsedQuery, config.params)
+  }
+}
+
 function getAuthMiddleware (secret) {
   return (config) => {
-    const { query, pathname } = url.parse(config.url, true)
-
-    const jwtToken = jwt.encode({
-      ...getExpirationInSeconds(),
-      iss,
-      qsh: jwt.createQueryStringHash({
-        method: config.method,
-        originalUrl: pathname,
-        query
-      })
-    }, secret)
-
-    return {
-      ...config,
-      headers: {
-        ...config.headers,
-        Authorization: `JWT ${jwtToken}`
-      }
-    }
+    const qsh = jwt.createQueryStringHash(jwt.reqFromAxiosConfig(config))
+    const jwtToken = jwt.encode({ ...getExpirationInSeconds(), iss, qsh }, secret)
```

Comment:
I suggest using more explicit names instead of abbreviations. looking up `qsh` forced me to read the line above and I still don't know what `iss` is 😆 .
I know, I am lazy, but I would love to hear your thoughts.

`Edit` ... found it at the top of the file. Not sure if it is worth assuming jwt API knowledge just to have the object property shorthand syntax... 🤔